### DISCUSSION TITLE
Maybe default value of table property `write.upsert.enabled` should be true

### DISCUSSION THREAD

**Discussion by baiyangtx** [apache/amoro#956] [General]
Currently, the spark batch writes and the flink streaming write will use the same table property to guarantee the uniqueness of the primary key. And for spark writes, the uniqueness of the primary key should be guaranteed by default,  on the other hand, I think the uniqueness of the primary key should be guaranteed at a high level, no matter whether in batch or streaming write so I think the default value for `write.upsert.enabled` should be true.

**Comment by baiyangtx** [comment]
@zhoujinsong @YesOrNo828 CC

**Comment by YesOrNo828** [comment]
Before changing the default value of the `write.upsert.enabled` to `true`, I think we should address this issue first #964 .
